File: Program 3/DictionaryTest.java
//Jeffrey Yeung
//jeyyeung
//CMPS 12B
//February 3, 2018
//Dictionary test file

public class DictionaryTest {
  public static void main(String[] args) {
    Dictionary A = new Dictionary();

    //testing isEmpty()
    System.out.println("testing isEmpty()");
    System.out.println("True or False, the dictionary is empty: " + A.isEmpty()); //return true
    System.out.println();

    //testing size()
    System.out.println("testing size()");
    System.out.println("The size of the dictionary is: " + A.size()); //return 0
    System.out.println();

    //testing insert()
    System.out.println("testing insert(), testing isEmpty(), testing size()");
    A.insert("1","z");
    A.insert("2","y");
    A.insert("3","x");
    A.insert("4","w");
    A.insert("5","v");
    A.insert("6","a");
    A.insert("7","b");
    A.insert("8","c");
    //A.insert("6","d"); //will cause a duplicate key exception
    System.out.println(A); //return 8 keys and values
    System.out.println("True or False, the dictionary is empty: " + A.isEmpty()); //return false
    System.out.println("The size of the dictionary is: " + A.size()); //return 8
    System.out.println();

    //testing lookup()
    String v;
    System.out.println("testing lookup()");
    v = A.lookup("1");
    System.out.println("key=1 "+(v==null?"not found":("value="+v))); //return key=1 value=z
    v = A.lookup("6");
    System.out.println("key=6 "+(v==null?"not found":("value="+v))); //return key=6 value=a
    v = A.lookup("20");
    System.out.println("key=20 "+(v==null?"not found":("value="+v))); //return key=20 not found
    System.out.println();

    //testing delete()
    System.out.println("testing delete(), testing isEmpty(), testing size()");
    A.delete("2");
    A.delete("4");
    A.delete("6");
    //A.delete("20"); //key not found exception
    System.out.println(A); //return 5 keys and values
    System.out.println("True or False, the dictionary is empty: " + A.isEmpty()); //return false
    System.out.println("The size of the dictionary is: " + A.size()); //return 5
    System.out.println();

    //testing makeEmpty()
    System.out.println("test makeEmpty(), testing isEmpty(), testing size()");
    System.out.println("True or False, the dictionary is empty: " + A.isEmpty()); //confirms dictionary is not empty
    System.out.println("The size of the dictionary is: " + A.size()); //confirms dictionary size
    A.makeEmpty(); //makes the dictionary empty
    System.out.println("True or False, the dictionary is empty: " + A.isEmpty()); //returns true
    System.out.println("The size of the dictionary is: " + A.size()); //returns 0
    System.out.println(A); //nothing to print since dictionary is empty

  }
}

File: Program 5/DictionaryTest.c
//Jeffrey Yeung
//jeyyeung
//CMPS 12B
//March 12, 2018
//test file for Dictionary.c

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){
   Dictionary A = newDictionary();
   char* k;
   char* v;
   char* word1[] = {"one","two","three","four","five","six"};
   char* word2[] = {"eleven","twelve","thirteen","fourteen","fifteen","sixteen"};
   int i;

   printf("%s\n", (isEmpty(A)?"true":"false")); //true
   printf("%d\n", size(A)); //0, since its empty

   for(i = 0; i < 6; i++){
      insert(A, word1[i], word2[i]);
   }

   printDictionary(stdout, A);
   printf("%s\n", (isEmpty(A)?"true":"false")); //false, inputed 6 new elements
   printf("%d\n", size(A)); //size: 6

   for(i = 0; i < 6; i++){
      k = word1[i];
      v = lookup(A, k);
      printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
   }

   //insert(A, "two", "twelve"); // error: duplicate keys

   delete(A, "one");
   delete(A, "four");
   delete(A, "six");

   printDictionary(stdout, A);

   for(i = 0; i < 6; i++){
      k = word1[i];
      v = lookup(A, k);
      printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
   }

   //delete(A, "five");  // error: key not found

   printf("%s\n", (isEmpty(A)?"true":"false")); //false
   printf("%d\n", size(A)); //3
   makeEmpty(A);
   printf("%s\n", (isEmpty(A)?"true":"false")); //true

   freeDictionary(&A);

   return(EXIT_SUCCESS);
}
